R/addin.R

```diff
@@ -69,29 +70,47 @@ confl_create_post_from_Rmd <- function(Rmd_file = NULL, interactive = NULL,
     env = globalenv()
   )
 
+  # set confl setting
   front_matter <- rmarkdown::yaml_front_matter(Rmd_file, "UTF-8")
 
-  if (!is.null(title)) {
-    front_matter$title <- title
+  confl_setting <- purrr::list_modify(front_matter$confl_setting, ...)
+
+  if (is.null(confl_setting$title)) {
+    confl_setting$title <- front_matter$title
+  }
```

Comment:
Good catch! `conflr_settings` probably doesn't contain `title`, so I think we always need this here. But, this line should be

``` r
  conflr_settings$title <- title %||% front_matter$title
```

which means

1. Use `title` argument if it's not `NULL`
2. Otherwise, use `title` parameter in the front matter
3. `title` is ignored if it's in `conflr_settings`; I cannot imagine a case where we need to handle multiple `title` parameters like:

```
---
title: Title1
conflr_settings:
  title: Title2
---

Reply:
OK, I will modify it.